tests/integ/test_tuner.py

```diff
@@ -31,70 +32,258 @@
+    kmeans.epochs = 1
 
-        tuner.fit([records, test_records])
+    return kmeans
 
-        print('Started hyperparameter tuning job with name:' + tuner.latest_tuning_job.name)
 
-        time.sleep(15)
-        tuner.wait()
+@pytest.fixture(scope='module')
+def hyperparameter_ranges():
+    return {'extra_center_factor': IntegerParameter(1, 10),
+            'mini_batch_size': IntegerParameter(10, 100),
+            'epochs': IntegerParameter(1, 2),
+            'init_method': CategoricalParameter(['kmeans++', 'random'])}
+
+
+def _tune_and_deploy(kmeans_estimator, kmeans_train_set, sagemaker_session, hyperparameter_ranges=None, job_name=None,
+                     warm_start_config=None):
+    tuner = _tune(kmeans_estimator, kmeans_train_set,
+                  hyperparameter_ranges=hyperparameter_ranges, warm_start_config=warm_start_config, job_name=job_name)
+    _deploy(kmeans_train_set, sagemaker_session, tuner)
 
+
+def _deploy(kmeans_train_set, sagemaker_session, tuner):
     best_training_job = tuner.best_training_job()
     with timeout_and_delete_endpoint_by_name(best_training_job, sagemaker_session):
         predictor = tuner.deploy(1, 'ml.c4.xlarge')
-        result = predictor.predict(train_set[0][:10])
+
+        result = predictor.predict(kmeans_train_set[0][:10])
 
         assert len(result) == 10
         for record in result:
             assert record.label['closest_cluster'] is not None
             assert record.label['distance_to_cluster'] is not None
 
 
+def _tune(kmeans_estimator, kmeans_train_set, tuner=None,
+          hyperparameter_ranges=None, job_name=None, warm_start_config=None,
+          wait_till_terminal=True, max_jobs=2, max_parallel_jobs=2):
+    with timeout(minutes=TUNING_DEFAULT_TIMEOUT_MINUTES):
+
+        if not tuner:
+            tuner = HyperparameterTuner(estimator=kmeans_estimator, objective_metric_name='test:msd',
+                                        hyperparameter_ranges=hyperparameter_ranges, objective_type='Minimize',
+                                        max_jobs=max_jobs,
+                                        max_parallel_jobs=max_parallel_jobs, warm_start_config=warm_start_config)
+
+        records = kmeans_estimator.record_set(kmeans_train_set[0][:100])
+        test_record_set = kmeans_estimator.record_set(kmeans_train_set[0][:100], channel='test')
+
+        tuner.fit([records, test_record_set], job_name=job_name)
+        print('Started hyperparameter tuning job with name:' + tuner.latest_tuning_job.name)
+
+        if wait_till_terminal:
+            tuner.wait()
+
+    return tuner
+
+
+@pytest.mark.continuous_testing
+def test_tuning_kmeans(sagemaker_session,
+                       kmeans_train_set,
+                       kmeans_estimator,
+                       hyperparameter_ranges):
+    _tune_and_deploy(kmeans_estimator, kmeans_train_set, sagemaker_session, hyperparameter_ranges=hyperparameter_ranges)
+
+
+@pytest.mark.continuous_testing
```

Comment:
how many new tests are we adding to the continuous test suite? are they all necessary?

Reply:
Only 2 `test_create_transfer_learning_tuner` and `test_create_tuning_kmeans_identical_dataset_algorithm_tuner` covering 2 different types of jobs... then there is a negative test which is just executed for single digit second and I would not worry about that.